micronaut-mcp-server-java-sdk/src/test/java/io/micronaut/mcp/server/stdio/async/AsyncResourcesTest.java

```diff
@@ -101,50 +157,68 @@ public void close() throws IOException {
             String uriTemplate = "pgn://round/{round}";
             String name = "2024ChessChampionshipRoundPgn";
             String title = "PGN of a round World Chess Championship 2024";
             String description = "Given a round, it returns a PGN of the World Chess Championship 2024 between Ding Liren and Gukesh Dommaraju";
-            return new McpSchema.ResourceTemplate(uriTemplate, name, title, description, PGN_MIME_TYPE, null, null);
+            return new McpServerFeatures.AsyncResourceTemplateSpecification(
+                new McpSchema.ResourceTemplate(uriTemplate, name, title, description, PGN_MIME_TYPE, null, null),
+                (mcpTransportContext, readResourceRequest) -> {
+                    Integer round = round(readResourceRequest.uri());
+                    return Mono.just(readResourceResult(readResourceRequest.uri(), round));
+                });
         }
 
-        @EachBean(McpSchema.Resource.class)
+        @EachBean(PgnFile.class)
         @Singleton
-        McpServerFeatures.AsyncResourceSpecification createPgnSyncResourceSpecification(McpSchema.Resource resource) {
+        McpServerFeatures.AsyncResourceSpecification createPgnSyncResourceSpecification(PgnFile pgnFile) {
+            McpSchema.Resource resource = getResource(pgnFile);
             return new McpServerFeatures.AsyncResourceSpecification(resource, (mcpSyncServerExchange, readResourceRequest) -> {
                 String uri = readResourceRequest.uri();
-                int lastSlash = uri.lastIndexOf('/');
-                String roundStr = uri.substring(lastSlash + 1);
-                int round = Integer.parseInt(roundStr);
-                List<McpSchema.ResourceContents> contents = new ArrayList<>();
-                pgnLoader.loadPgn(round).ifPresent(text ->
-                    contents.add(new McpSchema.TextResourceContents(uri, PGN_MIME_TYPE, text)));
-                return Mono.just(new McpSchema.ReadResourceResult(contents));
+                Integer round = round(readResourceRequest.uri());
+                return Mono.just(readResourceResult(uri, round));
             });
         }
+
+        private Integer round(String uri) {
+            int lastSlash = uri.lastIndexOf('/');
+            String roundStr = uri.substring(lastSlash + 1);
+            return Integer.parseInt(roundStr);
+        }
+
+        private McpSchema.ReadResourceResult readResourceResult(String uri, Integer round) {
+            List<McpSchema.ResourceContents> contents = new ArrayList<>();
+            pgnLoader.loadPgn(round).ifPresent(text ->
+                contents.add(new McpSchema.TextResourceContents(uri, PGN_MIME_TYPE, text)));
+            return new McpSchema.ReadResourceResult(contents);
+        }
+
+        private McpSchema.Resource getResource(PgnFile pgnFile) {
+            Optional<InputStream> roundPgnInputStreamOptional = resourceLoader.getResourceAsStream(pgnFile.getPath());
```

Comment:
where is this input stream closed?

Reply:
resolved via https://github.com/micronaut-projects/micronaut-mcp/pull/70/commits/976b1f28f6a18e63738cac1b94864a60a40972f0